src/substrait/textplan/StructuredSymbolData.h

```diff
@@ -37,6 +37,20 @@ struct RelationData {
   // Column name for each field known to this relation (in field order).  Used
   // to determine what fields are coming in as well and fields are going out.
   std::vector<const SymbolInfo*> fieldReferences;
+
+  // Each field reference here was generated within the current relation.
+  std::vector<const SymbolInfo*> generatedFieldReferences;
+
+  // Local aliases for field references in this relation.
+  std::map<size_t, std::string> generatedFieldReferenceAlternativeExpression;
```

Comment:
Should we, in a comment, give a brief example where this might be needed?

Reply:
Done.